app/js/components/Workflows.jsx

```diff
@@ -2,24 +2,22 @@ import React from 'react';
 
 import Workflow from './Workflow';
 
-const Workflows = ({ plugin }, { router }) => (
+const Workflows = ({ plugin, openWorkflow }) => (
     <div>
         { plugin.workflows.map(workflow => (
             <Workflow
                 key={workflow.name}
                 flow={workflow}
-                onClick = {() => router.push(`job/${plugin.name}/${workflow.name}`)}
+                onClick = {() => openWorkflow(plugin, workflow)}
             />
         ))}
     </div>
 );
 
 
 Workflows.propTypes = {
-    plugin: React.PropTypes.object
-};
-
-Workflows.contextTypes = {
+    openWorkflow: React.PropTypes.func,
+    plugin: React.PropTypes.object,
     router: React.PropTypes.object
```

Comment:
Part of me is tempted to switch to `Flow` and skip using `propTypes/contextTypes` they are just a runtime typecheck with relatively limited range.